cpp/src/arrow/buffer.h

```diff
@@ -346,6 +346,8 @@ class ARROW_EXPORT Buffer {
   static Result<std::shared_ptr<Buffer>> ViewOrCopy(
       std::shared_ptr<Buffer> source, const std::shared_ptr<MemoryManager>& to);
 
+  virtual std::shared_ptr<DeviceSync> get_device_sync() { return nullptr; }
```

Comment:
This name is somewhat ambiguous. Is this a sync event? Maybe `get_device_sync_event`?

Reply:
My vote for the bikeshed is:
```
// nest the class inside Device
DeviceSync -> Device::SyncEvent

// for trivial property accessors we don't usually prefix `get_`
get_device_sync -> device_sync_event
           (or) -> GetDeviceSyncEvent
```